File: src/base/_Number.ts
export class _Number extends Number {
  // 进制转换
  static convertBase(x, { from = 10, to = 10 } = {}): string {
    return Number.parseInt(x, from).toString(to);
  }
  // 判断素数
  static isPrime(x): boolean {
    if (x <= 1) {
      return false;
    }
    if (x === 2) {
      return true;
    }
    if (x % 2 === 0) {
      return false;
    }
    for (let i = 3; i <= Math.sqrt(x); i += 2) {
      if (x % i === 0) {
        return false;
      }
    }
    return true;
  }

  constructor(value) {
    value = Number.parseFloat(value);
    super(value);
  }

  // 相对于 Number.prototype.toFixed，移除尾部多余的零和小数点，以精简显示
  toMaxFixed(fractionDigits = 0): string {
    const str = Number.prototype.toFixed.call(this, fractionDigits);
    // 移除尾部多余的零和小数点
    return str.replace(/(\.\d*?[1-9])0+$|\.0*$/, '$1');
  }

  // 转换系列方法：转换成原始值或其他类型
  [Symbol.toPrimitive](hint: string) {
    if (hint === 'number') {
      return this.toNumber();
    }
    if (hint === 'string' || hint === 'default') {
      return this.toString();
    }
    return null;
  }
  toNumber() {
    // @ts-ignore
    return Number.parseFloat(this);
  }
  toString() {
    return String(this);
  }
  toBoolean() {
    // @ts-ignore
    return ![0, NaN].includes(this);
  }
  toJSON() {
    return this.toNumber();
  }
  toBigInt() {
    return BigInt(this.toNumber());
  }
}

File: python/src/base/_Number.py
from .es import Number, Math, parseInt
from ._Object import _Object

class _Number(Number):
  @staticmethod
  def convertBase(x, options = { }, _from = 10, to = 10) -> str:
    options = _Object(options)
    _from = options['from'] or _from
    to = options['to'] or to

    x = parseInt(x, _from)
    if to == 2:
      return format(x, 'b')
    elif to == 8:
      return format(x, 'o')
    elif to == 16:
      return format(x, 'x')
    else:
      # 对于十进制以外的其他进制，需要手动转换
      result = ''
      while x > 0:
        remainder = x % to
        result = str(remainder) + result
        x = x // to
      return result or '0'
  @staticmethod
  def isPrime(x) -> bool:
    if x <= 1:
      return False
    if x == 2:
      return True
    if x % 2 == 0:
      return False
    for i in range(3, parseInt(Math.sqrt(x)) + 1, 2):
      if x % i == 0:
        return False
    return True

  def toMaxFixed(x, fractionDigits = 0) -> str:
    # 创建格式化字符串
    format_str = f"{{:.{fractionDigits}f}}"
    # 格式化数字
    str_x = format_str.format(x)
    # 移除尾部多余的零和小数点
    return str_x.rstrip('0').rstrip('.') if '.' in str_x else str_x

__all__ = ['_Number']
